x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportRevertModelSnapshotAction.java

```diff
@@ -176,6 +176,13 @@ private void isBlocked(Job job, ActionListener<Boolean> listener) {
             // in order to complete and eventually unblock the job.
             GetTaskRequest getTaskRequest = new GetTaskRequest();
             getTaskRequest.setTaskId(job.getBlocked().getTaskId());
+
+            // If the request is forced, we will also wait for the existing task to finish
+            // to give a chance to this request to be executed without returning an error.
+            // This is particularly useful when a relocating job is calling revert.
+            getTaskRequest.setWaitForCompletion(request.isForce());
+            getTaskRequest.setTimeout(request.masterNodeTimeout());
```

Comment:
I think this timeout needs to be longer _if_ we are waiting for completion.  We revert with `delete_intervening_results=true`, and that could take a very long time if there are lots of results.  Maybe the timeout could be `MachineLearningField.STATE_PERSIST_RESTORE_TIMEOUT`.

But if we are not waiting for completion then the default timeout should be fine.

Reply:
We do set the timeout to a large value. In `OpenJobPersistentTaskExecutor.revertToCurrentSnapshot` we set:

```
RevertModelSnapshotAction.Request request = new RevertModelSnapshotAction.Request(jobId,
    jobSnapshotId == null ? ModelSnapshot.EMPTY_SNAPSHOT_ID : jobSnapshotId);
request.setForce(true);
request.setDeleteInterveningResults(true);
request.masterNodeTimeout(PERSISTENT_TASK_MASTER_NODE_TIMEOUT);
```

Reply:
Oh, I see, it's copying the long timeout from the original request to the wait for completion.

I think that's quite unintuitive though, because usually master node timeout is different to just "timeout".

Maybe you can at least add a comment explaining what's going on here: if the revert comes directly from an end user then the master node timeout is likely to be 30 seconds and if it comes from the persistent task executor then will be several days.  So we get the behaviour we want but not in a particular obvious or natural way.

Reply:
Ah, I see the confusion with master and non-master timeout. Let me see if I can make sense of that.